source/common/protobuf/utility.cc

```diff
@@ -306,6 +337,57 @@ void MessageUtil::checkForUnexpectedFields(const Protobuf::Message& message,
 
 namespace {
 
+// A proto visitor that validates the correctness of google.protobuf.Duration messages
+// as defined by Envoy's duration constraints.
+class DurationFieldProtoVisitor : public ProtobufMessage::ConstProtoVisitor {
+public:
+  void onField(const Protobuf::Message& message, const Protobuf::FieldDescriptor& field) override {
```

Comment:
The validation logic looks fine, but I am not sure why you do it in the field visitor and not message visitor? Does `onMessage` not trigger on field values?

Reply:
The way I looked at it, is validating a field of type Duration. I guess that the same can be with done with a message of the same type.
Not sure what makes more sense, and I'm open to trying to convert to a message-validator if you think it improves readability (or anything else).

Reply:
I just gave it another look, and you are correct, it is also much simpler when using `onMessage()`.
Thanks for making everyone's lives easier :)